main/src/cgeo/geocaching/MainActivity.java

```diff
@@ -195,21 +202,29 @@ public void accept(final Status gpsStatus) {
         }
     };
 
-    private final Handler firstLoginHandler = new Handler() {
+    private static final class FirstLoginHandler extends Handler {
+        private final WeakReference<MainActivity> activityRef;
```

Comment:
That's not possible, because `MainActivity` does not extend `Activity`, which is expected by `WeakReferenceHandler`.